File: backend/src/main/java/com/attendance/domain/AttendanceRecord.java
package com.attendance.domain;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;
import java.time.LocalDate;

@Entity
@Table(name = "attendance_records",
       uniqueConstraints = @UniqueConstraint(columnNames = {"user_id", "work_date"}))
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AttendanceRecord {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "work_date", nullable = false)
    private LocalDate workDate;

    @Column(name = "clock_in_time")
    private Instant clockInTime;

    @Column(name = "clock_out_time")
    private Instant clockOutTime;

    @Column(name = "is_late", nullable = false)
    @Builder.Default
    private boolean isLate = false;

    @Column(name = "late_minutes", nullable = false)
    @Builder.Default
    private int lateMinutes = 0;

    @Column(name = "is_early_leave", nullable = false)
    @Builder.Default
    private boolean isEarlyLeave = false;

    @Column(name = "short_minutes", nullable = false)
    @Builder.Default
    private int shortMinutes = 0;

    @Column(name = "notification_sent", nullable = false)
    @Builder.Default
    private boolean notificationSent = false;

    @Column(name = "created_at", updatable = false)
    @Builder.Default
    private Instant createdAt = Instant.now();

    @Column(name = "updated_at")
    @Builder.Default
    private Instant updatedAt = Instant.now();

    @PreUpdate
    void onUpdate() { this.updatedAt = Instant.now(); }
}

File: flutter/lib/core/models/attendance_record.dart
class AttendanceRecord {
  const AttendanceRecord({
    required this.id,
    required this.userId,
    required this.userFullName,
    required this.workDate,
    required this.clockInTime,
    required this.clockOutTime,
    required this.isLate,
    required this.lateMinutes,
    required this.isEarlyLeave,
    required this.shortMinutes,
    required this.workDurationMinutes,
  });

  final int id;
  final int userId;
  final String userFullName;
  final String workDate;
  final String? clockInTime;
  final String? clockOutTime;
  final bool isLate;
  final int lateMinutes;
  final bool isEarlyLeave;
  final int shortMinutes;
  final int? workDurationMinutes;

  factory AttendanceRecord.fromJson(Map<String, dynamic> json) {
    return AttendanceRecord(
      id: json['id'] as int? ?? 0,
      userId: json['userId'] as int? ?? 0,
      userFullName: json['userFullName'] as String? ?? '',
      workDate: json['workDate'] as String? ?? '',
      clockInTime: json['clockInTime'] as String?,
      clockOutTime: json['clockOutTime'] as String?,
      isLate: json['isLate'] as bool? ?? false,
      lateMinutes: json['lateMinutes'] as int? ?? 0,
      isEarlyLeave: json['isEarlyLeave'] as bool? ?? false,
      shortMinutes: json['shortMinutes'] as int? ?? 0,
      workDurationMinutes: json['workDurationMinutes'] as int?,
    );
  }
}